controllers/storagecluster/cephcluster.go

```diff
@@ -1180,38 +1203,64 @@ func applyLabels(labels map[string]string, t *metav1.ObjectMeta) {
+type replaceToken struct {
 	groupName         string
 	recordOrAlertName string
-	wordInExpr        string
+	wordToReplace     string
 	replaceWith       string
 }
 
-func changePromRuleExpr(promRules *monitoringv1.PrometheusRule, replaceTokens []exprReplaceToken) {
-	if promRules == nil {
+func createReplaceToken(groupName, recordOrAlertName, wordToReplace, replaceWith string) replaceToken {
+	return replaceToken{
+		groupName:         groupName,
+		recordOrAlertName: recordOrAlertName,
+		wordToReplace:     wordToReplace,
+		replaceWith:       replaceWith,
+	}
+}
+
+// changePromRule replaces the wordToReplace with replaceWith in the PrometheusRule
+// This can be used to update the values in the PrometheusRule dynamically
+func changePromRule(promRule *monitoringv1.PrometheusRule, tokens []replaceToken) {
+	if promRule == nil {
 		return
 	}
-	for _, eachToken := range replaceTokens {
-		// if both the words, one being replaced and the one replacing it, are same
-		// then we don't have to do anything
-		if eachToken.replaceWith == eachToken.wordInExpr {
+
+	// Iterate over each token for replacements
+	for _, token := range tokens {
+		// Skip if the word and replacement are the same
+		if token.replaceWith == token.wordToReplace {
 			continue
 		}
-		for gIndx, currGroup := range promRules.Spec.Groups {
-			if eachToken.groupName != "" && eachToken.groupName != currGroup.Name {
+
+		// Iterate through all groups in the Prometheus rule
+		for groupIdx := range promRule.Spec.Groups {
+			group := &promRule.Spec.Groups[groupIdx]
+			// If groupName is specified, ensure it matches; otherwise, apply to all groups
+			if token.groupName != "" && token.groupName != group.Name {
 				continue
 			}
-			for rIndx, currRule := range currGroup.Rules {
-				if eachToken.recordOrAlertName != "" {
-					if currRule.Record != "" && currRule.Record != eachToken.recordOrAlertName {
-						continue
-					} else if currRule.Alert != "" && currRule.Alert != eachToken.recordOrAlertName {
-						continue
+
+			// Iterate through the rules in the group
+			for ruleIdx := range group.Rules {
+				rule := &group.Rules[ruleIdx]
+				// If recordOrAlertName is specified, ensure it matches; otherwise, apply to all rules
+				if token.recordOrAlertName == "" || rule.Record == token.recordOrAlertName || rule.Alert == token.recordOrAlertName {
+					// Update the annotations in the rule
+					if rule.Annotations != nil {
+						// Update description if it exists
+						if description, exists := rule.Annotations["description"]; exists {
+							newDescription := strings.Replace(description, token.wordToReplace, token.replaceWith, -1)
+							rule.Annotations["description"] = newDescription
+						}
+					}
+					// Update the expression field in the rule
+					exprStr := rule.Expr.String()
+					if exprStr != "" {
+						newExpr := strings.Replace(exprStr, token.wordToReplace, token.replaceWith, -1)
+						rule.Expr = intstr.Parse(newExpr)
 					}
 				}
```

Comment:
If we apply the `RuleSection` (and other needed changes), `changePromRule()` function can be cleaned out as follows,

```suggestion
			for ruleIdx, rule := range group.Rules {
				// If recordOrAlertName is specified, ensure it matches; otherwise, apply to all rules
				if token.recordOrAlertName == "" || rule.Record == token.recordOrAlertName || rule.Alert == token.recordOrAlertName {
					switch token.wordInSection {
					case ExprRuleSection:
						// Update the expression field in the rule
						newExpr := strings.ReplaceAll(rule.Expr.String(), token.wordToReplace, token.replaceWith)
						promRule.Spec.Groups[groupIdx].Rules[ruleIdx].Expr = intstr.Parse(newExpr)
					case AnnotationRuleSection:
						if rule.Annotations == nil { // if no annotations, continue
							continue
						}
						annotMap := rule.Annotations
						annotMap[token.sectionKey] = strings.ReplaceAll(
							annotMap[token.sectionKey], token.wordToReplace, token.replaceWith)
					case LabelRuleSection:
						if rule.Labels == nil { // if there are no labels, continue
							continue
						}
						labelMap := rule.Labels
						labelMap[token.sectionKey] = strings.ReplaceAll(
							labelMap[token.sectionKey], token.wordToReplace, token.replaceWith)
					}
				}
			}
```